shared/src/map/layers/tiled/vector/symbol/SymbolAnimationCoordinatorMap.h

```diff
@@ -94,45 +104,52 @@ class SymbolAnimationCoordinatorMap {
-                ++nextSetIt;
-                for (auto coordsIt = setIt->second.begin(); coordsIt != setIt->second.end();) {
-                    if (!coordsIt->get()->isUsed()) {
-                        coordsIt = setIt->second.erase(coordsIt);
-                    } else {
-                        ++coordsIt;
+        for (auto &shard : shards) {
+            std::lock_guard<std::mutex> lock(shard.mutex);
+            for (auto it = shard.animationCoordinators.begin(), next_it = it; it != shard.animationCoordinators.end(); it = next_it) {
+                ++next_it;
+                for (auto setIt = it->second.begin(), nextSetIt = setIt; setIt != it->second.end(); setIt = nextSetIt) {
+                    ++nextSetIt;
+                    for (auto coordsIt = setIt->second.begin(); coordsIt != setIt->second.end();) {
+                        if (!coordsIt->get()->isUsed()) {
+                            coordsIt = setIt->second.erase(coordsIt);
+                        } else {
+                            ++coordsIt;
+                        }
+                    }
+                    if (setIt->second.empty()) {
+                        it->second.erase(setIt);
                     }
                 }
-                if (setIt->second.empty()) {
-                    it->second.erase(setIt);
+                if (it->second.empty()) {
+                    shard.animationCoordinators.erase(it);
                 }
             }
-            if (it->second.empty()) {
-                animationCoordinators.erase(it);
-            }
         }
     }
 
     void enableAnimations(bool enabled) {
         animationsEnabled = enabled;
-        std::lock_guard<std::mutex> lock(mapMutex);
-        for (const auto &[_, coordinatorMap] : animationCoordinators) {
-            for (const auto &[_, coordinatorSet] : coordinatorMap) {
-                for (const auto &coordinator : coordinatorSet) {
-                    coordinator->enableAnimations(enabled);
+        for (auto &shard : shards) {
+            std::lock_guard<std::mutex> lock(shard.mutex);
+            for (const auto &[_, coordinatorMap] : shard.animationCoordinators) {
+                for (const auto &[_, coordinatorSet] : coordinatorMap) {
+                    for (const auto &coordinator : coordinatorSet) {
+                        coordinator->enableAnimations(enabled);
+                    }
                 }
             }
         }
     }
 
 private:
-    static const size_t MIN_NUM_SEARCH = 9;
+    struct Shard {
+        std::mutex mutex;
+        std::unordered_map<size_t, std::map<int, std::set<std::shared_ptr<SymbolAnimationCoordinator>, CoordinatorXCompare>>> animationCoordinators;
+    };
 
     bool animationsEnabled = true;
-
-    std::mutex mapMutex;
-    std::unordered_map<size_t, std::map<int, std::set<std::shared_ptr<SymbolAnimationCoordinator>, CoordinatorXCompare>>> animationCoordinators;
-};
+    size_t numShards;
+    std::vector<Shard> shards;
```

Comment:
Could make this `static constexpr numShards = 16` and `std::array<Shard, numShards>`, for one fewer indirection and allocation, and optimized division in the access.